Source/cmLinkDatabaseGenerator.cpp

```diff
@@ -0,0 +1,62 @@
+//
+// Created by Gleb Mirzazyanov on 31.05.2021.
+//
+
+#include "cmLinkDatabaseGenerator.h"
+
+void cmLinkDatabaseGenerator::GenerateLinkCommand(
+  const std::vector<std::string>& dependencyFiles,
+  const std::string& workingDirectory,
+  const std::vector<std::string>& linkCommands, const std::string& version,
+  bool isOnOption, std::set<int> ranlibIndexes, const std::string& buildFileDir)
+{
+
+  std::string linkCommandsFile = "/link_commands.json";
+
+  if (!LinkCommandsJson) {
+    std::string commandDatabaseName =
+      cmStrCat(buildFileDir, linkCommandsFile);
+
+    Json::Value jsonVersionValue;
+    if (!version.empty()) {
+      if(isOnOption){
+        jsonVersionValue["version"] = linkDatabaseRelease;
+      } else {
+        jsonVersionValue["version"] = version;
+      }
+    } else {
+      return;
+    }
+
+    LinkCommandStream =
+      cm::make_unique<cmGeneratedFileStream>(commandDatabaseName);
+    LinkCommandsJson.append(jsonVersionValue);
+  }
+
+  Json::Value linkJsonCommand;
+  linkJsonCommand["directory"] =
+    cmGlobalGenerator::EscapeJSON(workingDirectory);
+
+  Json::Value files(Json::arrayValue);
+  for (const auto& sourceFile : dependencyFiles) {
+    files.append(Json::Value(sourceFile));
```

Comment:
Do you need EscapeJson here? What if filename is with spaces inside? For example "hello world.cpp"? Will it work?